plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -41,19 +46,62 @@ func NewStrategyStore(options Options, logger *zap.Logger) (ss.StrategyStore, er
 		logger:            logger,
 		serviceStrategies: make(map[string]*sampling.SamplingStrategyResponse),
 	}
+
+	// Read strategies
 	strategies, err := loadStrategies(options.StrategiesFile)
 	if err != nil {
 		return nil, err
 	}
 	h.parseStrategies(strategies)
+
+	// Watch strategies file for changes.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Error("failed to create a new watcher for the sampling strategies file", zap.Error(err))
+	}
+
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
```

Comment:
Could you try this in Kubernetes with a file coming from a config map? Because of the two-level symlinking, I had some problems with the hot reload of the UI config (#1688).

Reply:
Does it make sense to have an e2e test suite in Jaeger to cover these scenarios?

Reply:
You mean, to test a Kubernetes-specific setup, or the nested symlinks ?

Reply:
A kubernetes-specific setup

Reply:
It might then belong to the operator repo, instead of this one.